scripts/bulk-access-control.md

```diff
@@ -0,0 +1,101 @@
+
+| Parameter                        | Description                                                                                                                                                                            |
+|----------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `-u`, <br>`--uuid`                | This parameter, mandatory and repeatable, specifies the uuid of the object to process according to processing instruction, specified in the json file (see below).                                                                                                              |
+
+| `-f`, <br>`--file`                | This parameter specifies the json file containing the processing instruction.                                                                                                              |
+| `-h`, <br>`--help`               | This parameter displays help information for the script.                                                                                                                               |
+
+The json file specified with the file parameter has the following structure
+ structured as follow
+```json
+{
+   item: {
+      mode: "replace",
+      accessConditions: [
+          {
+            "name": "openaccess"
+          }
+      ]
+   },
+   bitstream: {
+      constraints: {
+          uuid: [bit-uuid1, bit-uuid2, ..., bit-uuidN],
+      },
+      mode: "add",
+      accessConditions: [
+        {
+         "name": "embargo",
+         "startDate": "2024-06-24T23:59:59.999+0000"
+        },
+        {
+         "name": "network"         
+        }
+      ]
+   }
+}
+```
+* target: is a list of uuid of communities, collections and individual items to process
+* item: the item node contains the processing detail at the item level for the identified resources
+  * mode: can be replace or add. Replace will mean that after processing the items will have only the access conditions that are specified in the accessContitions property. Add will mean that after processing the items will have the specified accessCondition in addition to the ones already defined.
+  * accessConditions: the list of access conditions to apply, with their specific parameters if needed (such as a startDate for an embargo, etc.). Only in replace mode the accessConditions can be empty in this case the access conditions of the item will be reset to what is inherited from their owningCollection. Please note that when accessConditions is set and is not empty not inheritence from the owningCollection occur. For more information on accessConditions see the [What is an Access Condition?](#what-is-an-access-condition) section below.
+* bitstreams: the bitstream node contains the processing detail at the bitstream level for the identified resources
```

Comment:
We need to document that this only applies to bitstreams in the main content bundle (i.e. ORIGINAL bundle).  That's what the implementation does.  I'm OK with that approach (attempting to modify other bundles may be too complex), but it must be clearly documented.

Reply:
you are right we need to improve the documentation here. The requirement 
https://wiki.lyrasis.org/display/DSPACE/Bulk+access+control+management+-+4Science+proposal
requires that
> derived bitstreams created by the filter-media will be updated according to the changes performed to the corresponding original bitstream expect than for bitstreams derived by public filters classes that will stay with their original policies (set to READ Anonymous at the creation time)

This to assure that if you make a bitstream restricted and the thumbnail is configured to follow the same policies of the original bitstream also the thumbnail becomes restricted, and viceversa when you switch to open access.
It should be not too much complicated to respect this requirement that would be a huge benefit for administrators